### ISSUE TITLE
Shape problem when training Inception-BN model.

### ISSUE THREAD

**Issue by noirblack** [kyu-sz/WPAL-network#5]
I try to replace the blvc_googlenet with inception-BN model (https://github.com/pertusa/InceptionBN-21K-for-Caffe)
However, I encounter Check failed in loading snapshot.
"
I0417 19:20:23.298787  1975 net.cpp:283] Network initialization done.
I0417 19:20:23.299363  1975 solver.cpp:60] Solver scaffolding done.
Loading snapshot weights from data/pretrained/Inception21k.caffemodel
I0417 19:20:24.088156  1975 net.cpp:761] Ignoring source layer data
I0417 19:20:24.104071  1975 net.cpp:761] Ignoring source layer global_pool
I0417 19:20:24.104101  1975 net.cpp:761] Ignoring source layer fc1
I0417 19:20:24.104110  1975 net.cpp:761] Ignoring source layer softmax
Solving...
BlobFetcher started
F0417 19:20:25.426437  1975 concat_layer.cpp:42]  Check failed: top_shape[j] == bottom[i]->shape(j) (9 vs. 8) All inputs must have the same shape, except at concat_axis.
"
It seems that this is caused by the conv and pool layer have different formula for output size.
I try to set the TRAIN.SCALSE, but it did not work.
"./experiments/scripts/wpal_net.sh 0 INCEPTIONV2_RAP data/pretrained/Inception21k.caffemodel RAP 1 --set TRAIN.BATCH_SIZE 16 TRAIN.SCALES "[512]"

I can not figure out why there is top_shape like "9" and "8", for the reason that the minimize output shape should be "16" for 512 input.

**Comment by kyu-sz** [comment]
Some models contain inception structures that do not support varying input sizes.

As a workaround, you can make modification to them: e.g. replace the "conv_4e_double_3x3_1" layer, which has param:
```
  convolution_param { 
     num_output: 256
     kernel_size: 3
     stride: 2
     pad: 1
  }
```
with a combination of a convolution layer with the same name and param:

```
  convolution_param { 
     num_output: 256
     kernel_size: 3
     stride: 1
     pad: 1
  }
```
, and an extra pooling layer with param:
```
  pooling_param {
     pool: MAX
     kernel_size: 3
     stride: 2
     pad: 0
  }
```

This modification ensures the input-output size relationship of this branch remains the same with others, while not affecting the usage of the pre-trained model.
